addons/sms/wizard/sms_composer.py

```diff
@@ -233,11 +235,56 @@ def _action_send_sms_mass(self, records=None):
         records = records if records is not None else self._get_records()
 
         sms_record_values = self._prepare_mass_sms_values(records)
+
+        # create the messages for the chatter if logging
+        created_messages = self.env['mail.message']
+        if self.mass_keep_log and records and issubclass(type(records), self.pool['mail.thread']):
+            author_id, email_from = self.env['mail.thread']._message_compute_author(None, None, raise_on_email=False)
+            base_message_values = {
+                'subject': None,
+                'author_id': author_id,
+                'email_from': email_from,
+                'message_type': 'sms',
+                'model': records._name,
+                'subtype_id': self.env['ir.model.data']._xmlid_to_res_id('mail.mt_comment'),
+                'is_internal': False,
+                'reply_to': self.env['mail.thread']._notify_get_reply_to(default=email_from)[False],
+                'email_add_signature': False,
+            }
+
+            message_create_values = [dict(base_message_values,
+                                          body=sms_record_values[record.id]['body'],
+                                          res_id=record.id,
+                                          message_id=tools.generate_tracking_message_id(record.name),
+                                          ) for record in records]
+
+            created_messages = records.sudo()._message_create(message_create_values)
+
+            # update dictionaries to link the corresponding message, if any
+            record_to_message = {
+                message.res_id: message.id
+                for message in created_messages
+            }
+            for key, value in sms_record_values.items():
+                value.update({'mail_message_id': record_to_message[key]})
+
+        # create SMSes
         sms_all = self._prepare_mass_sms(records, sms_record_values)
 
-        if sms_all and self.mass_keep_log and records and issubclass(type(records), self.pool['mail.thread']):
-            log_values = self._prepare_mass_log_values(records, sms_record_values)
-            records._message_log_batch(**log_values)
+        if self.mass_notify:
+            # create notifications for tracking in chatter
+            notif_create_values = [{
```

Comment:
no need to create a variable `notif_create_values` -you can create inline directly)